tests/filters/test_filter_query.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Optional
+
+import pytest
+import strawberry
+import strawberry_django
+from django.db.models import Q
+from strawberry.arguments import UNSET
+import django_filters
+from tests import models
+
+
+@pytest.mark.django_db
+def test_filters_input():
+    group1 = models.Group.objects.create(name="group1")
+    tag1 = models.Tag.objects.create(name="tag1")
+    tag2 = models.Tag.objects.create(name="tag2")
+    models.User.objects.create(name="user1", group=group1, tag=tag1)
+    models.User.objects.create(name="user2", tag=tag2)
+    models.User.objects.create(name="user3")
+
+    @strawberry_django.create_filter
+    class UserFilter(django_filters.FilterSet):
+        name = django_filters.CharFilter(lookup_expr="icontains")
+        search = django_filters.CharFilter(method="filter_search")
+
+        def filter_search(self, queryset, name, value):
+            return queryset.filter(
+                Q(name__icontains=value) |
+                Q(group__name__icontains=value) |
+                Q(tag__name__icontains=value)
+            )
+
+        class Meta:
+            model = models.User
+            fields = ["name", "search"]
+
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def user_ids(self, filters: Optional[UserFilter] = UNSET) -> List[int]:
+            queryset = models.User.objects.all()
+            queryset = strawberry_django.apply_filter(filters, queryset)
+            return queryset.order_by("pk").values_list("pk", flat=True)
+
+    schema = strawberry.Schema(query=Query)
+    query = """
+        query getUserIds ($filters: UserFilter) {
+            userIds (filters: $filters)
+        }
+    """
+
+    # Execute without filter
+    variable_values = {}
+    result = schema.execute_sync(query, variable_values=variable_values)
+    assert result.data
+    assert result.data["userIds"] == [1, 2, 3]
+    assert result.errors == None
+
+    # Execute with empty filter
+    variable_values = {"filters": {}}
+    result = schema.execute_sync(query, variable_values=variable_values)
+    assert result.data
+    assert result.data.get("userIds", None) == [1, 2, 3]
+    assert result.errors == None
+
+    # Execute with None filter
+    variable_values = {"filters": None}
+    result = schema.execute_sync(query, variable_values=variable_values)
+    assert result.data
+    assert result.data.get("userIds", None) == [1, 2, 3]
+    assert result.errors == None
+
+    # Execute with name filter
+    variable_values = {"filters": {"name": "user1"}}
+    result = schema.execute_sync(query, variable_values=variable_values)
+    assert result.data
+    assert result.data.get("userIds", None) == [1]
+    assert result.errors == None
+
+    # Execute with search filter
```

Comment:
could we split these into smaller test cases so that one test would test one thing? pytest fixtures would help here

Reply:
Thanks for this test improvevent